engine/crates/engine-v2/schema/src/lib.rs

```diff
@@ -124,10 +126,23 @@ pub struct Object {
     pub description: Option<StringId>,
     pub interfaces: Vec<InterfaceId>,
     /// All directives that made it through composition. Notably includes `@tag`.
-    pub composed_directives: Vec<Directive>,
+    pub composed_directives: Directives,
```

Comment:
In my planning PR I've added `IdRange` in `crate::utils` for this exact purpose, so you should use `IdRange<DirectiveId>` here, same for the other ones. Avoids the need for the extra index and provides a way to iterate over the ids.

Reply:
Will do :+1: